From 711c07f017f50e6dc89434a92f4b1a65b8e896c9 Mon Sep 17 00:00:00 2001
From: Michael Wang <wangyun@linux.vnet.ibm.com>
Date: Mon, 8 Oct 2012 16:59:17 +0800
Subject: [PATCH] rcu: split 'rcuboost' to each flavor

commit 29c67764f121a0980eb30d0314821ea631e6cfaf upstream.

This patch add new 'rcuboost' to each flavor's folder, now we could use:
	'cat /debugfs/rcu/rsp/rcuboost'
to get the selected rsp info.

Signed-off-by: Michael Wang <wangyun@linux.vnet.ibm.com>
Signed-off-by: Paul E. McKenney <paulmck@linux.vnet.ibm.com>
Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
Signed-off-by: Yang Shi <yang.shi@windriver.com>

diff --git a/kernel/rcutree_trace.c b/kernel/rcutree_trace.c
index 65b6265..cae417d 100644
--- a/kernel/rcutree_trace.c
+++ b/kernel/rcutree_trace.c
@@ -241,7 +241,7 @@ static const struct file_operations rcu_node_boost_fops = {
 	.owner = THIS_MODULE,
 	.open = rcu_node_boost_open,
 	.read = seq_read,
-	.llseek = seq_lseek,
+	.llseek = no_llseek,
 	.release = single_release,
 };
 
@@ -459,6 +459,15 @@ static int __init rcutree_trace_init(void)
 					rspdir, rsp, &new_rcubarrier_fops);
 			if (!retval)
 				goto free_out;
+
+#ifdef CONFIG_RCU_BOOST
+			if (rsp == &rcu_preempt_state) {
+				retval = debugfs_create_file("rcuboost", 0444,
+					rspdir, NULL, &rcu_node_boost_fops);
+				if (!retval)
+					goto free_out;
+			}
+#endif
 	}
 
 	retval = debugfs_create_file("rcubarrier", 0444, rcudir,
-- 
1.7.5.4

